packages/react/slot/src/Slot.tsx

```diff
@@ -74,15 +74,19 @@ function mergeProps(slotProps: AnyProps, childProps: AnyProps) {
   // all child props should override
   const overrideProps = { ...childProps };
 
-  // if it's a handler, modify the override by composing the base handler
   for (const propName in childProps) {
     const slotPropValue = slotProps[propName];
     const childPropValue = childProps[propName];
-    const isHandler = /^on[A-Z]/.test(propName);
 
+    const isHandler = /^on[A-Z]/.test(propName);
+    // if it's a handler, modify the override by composing the base handler
     if (isHandler) {
       overrideProps[propName] = composeHandlers(childPropValue, slotPropValue);
     }
+    // if it's `style`, we merge them
+    else if (propName === 'style') {
+      overrideProps[propName] = { ...slotPropValue, ...childPropValue };
+    }
```

Comment:
This is probably the most notable change in this PR, had to make sure we compose `style` in `Slot` as well because we now have components that get slotted which have internal styles (`DismissableLayer`).